### ISSUE TITLE
image tile transformation in JiGen

### ISSUE THREAD

**Issue by smilesun** [marrlab/DomainLab#110]
The original author used the following transformation to each tile of the image, I am not sure why they/we need this? 

`https://github.com/marrlab/DomainLab/blob/f481c39468dc3ea32678dbac495ca959700c552b/domainlab/dsets/utils_wrapdset_patches.py#L12`

@Car-la-F , do you have some ideal if we need this or shall we remove this for a fair comparison?

**Comment by Car-la-F** [comment]
@smilesun,
I will look into this now. we can discuss about it in our meeting later.

**Comment by Car-la-F** [comment]
@smilesun,
I did some research and found the following:

_All pre-trained models expect input images normalized in the same way, i.e. mini-batches of 3-channel RGB images of shape (3 x H x W), where H and W are expected to be at least 224. The images have to be loaded in to a range of [0, 1] and then normalized using mean = [0.485, 0.456, 0.406] and std = [0.229, 0.224, 0.225]._
(https://pytorch.org/hub/pytorch_vision_alexnet/ first paragraph)

It seems like this normalisation is necessary when using alexnet, as the network expects such an input. Also other networks (like ResNet https://pytorch.org/hub/pytorch_vision_resnet/) seem to have the same property. It seems like it would be helpful to use this normalization for all of our nets which use this huge pretrained models.


My conclusion would be, it is **no cheating** and we should consider using this normalization whenever a pre-trained model is used which has this in the requirements.

**Comment by smilesun** [comment]
I think then we need the transformation be more generic instead of only for jigen, that means the permutated images should use the same transformation of the other algorithms.

**Comment by smilesun** [comment]
The issue is for a dataset, it also permit transformation itself:
https://github.com/marrlab/DomainLab/blob/37504b0d51b241076ab86374dfa5727bfec7b315/domainlab/dsets/dset_subfolder.py#L99
That means, if a transformation has been make already to the image, then we permute the tiles , do we need to do the same transformation again?

**Comment by Car-la-F** [comment]
@smilesun,
the permutations of the tiles in the image should not have any influence on the total mean and standard deviation of the pixel values. Hence, from my understanding, if we do a transformation before the permutation then the second transformation after the permutation is coincides with the identity function.

**Comment by schoersch** [comment]
Although the second normalization does nothing, it seems more consistent to me if none of the algorithms does some under-the-hood preprocessing, i.e. it is up to the dest_subfolder to transform and prepare the images.

**Comment by smilesun** [comment]
fixed in commit commit 410b051ad0ac80bdd8d6425a07626629f2d879be
